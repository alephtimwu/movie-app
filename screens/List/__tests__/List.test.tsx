import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
// import { CrpAll } from '__mocks__';
import GlobalState from '../../../store';
import HomeScreen from '../index';

const Stack = createStackNavigator();

const theme = {
  dark: true,
  colors: {
    primary: 'rgb(255, 45, 85)',
    background: '#1d1d27',
    card: 'rgb(255, 255, 255)',
    text: '#ffffff',
    border: 'rgb(199, 199, 204)'
  }
};

describe('HomeScreen =>', () => {
  let tree = null;
  const handlePress = jest.fn();

  beforeEach(() => {
    tree = render(
      <GlobalState.Provider>
        <NavigationContainer theme={theme}>
          <Stack.Navigator initialRouteName="Home">
            <Stack.Screen
              name="Home"
              component={HomeScreen}
              options={({ navigation, route }) => ({
                title: '',
                headerTransparent: true,
                headerStyle: {
                  backgroundColor: '#1D1D27',
                  borderWidth: 0,
                  borderColor: '#1D1D27'
                },
                headerTintColor: '#fff'
              })}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </GlobalState.Provider>
    );
  });

  test('HomeScreen should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  //   test('HomeScreen should render correctly with custom width height', () => {
  //     const { rerender, getByTestId } = tree;
  //     rerender(
  //       <HomeScreen
  //         movie={mockMovie.mock[0]}
  //         onPress={handlePress}
  //         width={80}
  //         height={100}
  //       />
  //     );
  //     fireEvent.press(getByTestId('card-press'));
  //     expect(handlePress).toBeCalled();
  //   });
});
