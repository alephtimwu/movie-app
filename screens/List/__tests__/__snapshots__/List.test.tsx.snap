// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen => HomeScreen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#1d1d27",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "backgroundColor": "#1d1d27",
                          "flex": 0,
                        }
                      }
                    />
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#1d1d27",
                            "borderBottomColor": "transparent",
                            "borderColor": "black",
                            "borderLeftColor": "transparent",
                            "borderRightColor": "transparent",
                            "borderStyle": "solid",
                            "borderTopColor": "transparent",
                            "borderWidth": 0,
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "center",
                              },
                              Object {
                                "justifyContent": "flex-start",
                              },
                              Object {
                                "marginLeft": 25,
                                "marginRight": 25,
                              },
                              Object {
                                "marginTop": 8,
                              },
                            ]
                          }
                        >
                          <View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                              theme={
                                Object {
                                  "colors": Object {
                                    "disabled": "hsl(208, 8%, 90%)",
                                    "divider": "#bcbbc1",
                                    "error": "#ff190c",
                                    "grey0": "#393e42",
                                    "grey1": "#43484d",
                                    "grey2": "#5e6977",
                                    "grey3": "#86939e",
                                    "grey4": "#bdc6cf",
                                    "grey5": "#e1e8ee",
                                    "greyOutline": "#bbb",
                                    "platform": Object {
                                      "android": Object {
                                        "error": "#f44336",
                                        "primary": "#2196f3",
                                        "secondary": "#9C27B0",
                                        "success": "#4caf50",
                                        "warning": "#ffeb3b",
                                      },
                                      "ios": Object {
                                        "error": "#ff3b30",
                                        "primary": "#007aff",
                                        "secondary": "#5856d6",
                                        "success": "#4cd964",
                                        "warning": "#ffcc00",
                                      },
                                    },
                                    "primary": "#2089dc",
                                    "searchBg": "#303337",
                                    "secondary": "#8F0CE8",
                                    "success": "#52c41a",
                                    "warning": "#faad14",
                                  },
                                }
                              }
                              underlayColor="white"
                            >
                              <Text />
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                                Object {
                                  "flex": 1,
                                },
                                Object {
                                  "justifyContent": "flex-end",
                                },
                              ]
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "marginRight": 20,
                                  "opacity": 1,
                                }
                              }
                            >
                              <View>
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "transparent",
                                      "justifyContent": "center",
                                    }
                                  }
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "disabled": "hsl(208, 8%, 90%)",
                                        "divider": "#bcbbc1",
                                        "error": "#ff190c",
                                        "grey0": "#393e42",
                                        "grey1": "#43484d",
                                        "grey2": "#5e6977",
                                        "grey3": "#86939e",
                                        "grey4": "#bdc6cf",
                                        "grey5": "#e1e8ee",
                                        "greyOutline": "#bbb",
                                        "platform": Object {
                                          "android": Object {
                                            "error": "#f44336",
                                            "primary": "#2196f3",
                                            "secondary": "#9C27B0",
                                            "success": "#4caf50",
                                            "warning": "#ffeb3b",
                                          },
                                          "ios": Object {
                                            "error": "#ff3b30",
                                            "primary": "#007aff",
                                            "secondary": "#5856d6",
                                            "success": "#4cd964",
                                            "warning": "#ffcc00",
                                          },
                                        },
                                        "primary": "#2089dc",
                                        "searchBg": "#303337",
                                        "secondary": "#8F0CE8",
                                        "success": "#52c41a",
                                        "warning": "#faad14",
                                      },
                                    }
                                  }
                                  underlayColor="white"
                                >
                                  <Text />
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "backgroundColor": "#fff",
                                    "borderRadius": 16,
                                    "height": 32,
                                    "width": 32,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#1D1D27",
                                    "fontSize": 10,
                                    "fontWeight": "bold",
                                    "textTransform": "uppercase",
                                  }
                                }
                              >
                                LVP
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              Object {
                                "flexDirection": "row",
                                "justifyContent": "space-around",
                                "padding": 10,
                              },
                            ]
                          }
                        >
                          <ActivityIndicator
                            animating={true}
                            color="#fff"
                            hidesWhenStopped={true}
                            size="large"
                          />
                        </View>
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 18,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
